contracts/src/v0.8/dev/ArbitrumSequencerStatusRecorder.sol

```diff
@@ -0,0 +1,245 @@
+import {FlagsInterface} from "./interfaces/FlagsInterface.sol";
+import {ArbitrumSequencerStatusRecorderInterface} from "./interfaces/ArbitrumSequencerStatusRecorderInterface.sol";
+import {SimpleReadAccessController} from "../SimpleReadAccessController.sol";
+import {ConfirmedOwner} from "../ConfirmedOwner.sol";
+
+/**
+ * @title ArbitrumSequencerStatusRecorder - L2 status history aggregator
+ * @notice L2 contract that receives status updates from a specific L1 address,
+ *  records a new answer if the status changed, and raises or lowers the flag on the
+ *   stored Flags contract.
+ */
+contract ArbitrumSequencerStatusRecorder is
+  AggregatorV2V3Interface,
+  ArbitrumSequencerStatusRecorderInterface,
+  TypeAndVersionInterface,
+  SimpleReadAccessController
+{
+  struct Round {
+    bool status;
+    uint64 timestamp;
+  }
+
+  event Initialized();
+  event L1OwnershipTransferred(address indexed from, address indexed to);
+
+  string private constant V3_NO_DATA_ERROR = "No data present";
+  /// @dev Follows: https://eips.ethereum.org/EIPS/eip-1967
+  address public constant FLAG_ARBITRUM_SEQ_OFFLINE =
+    address(bytes20(bytes32(uint256(keccak256("chainlink.flags.arbitrum-seq-offline")) - 1)));
+
+  uint8 public constant override decimals = 0;
+  string public constant override description = "L2 Status History";
+  uint256 public constant override version = 1;
+
+  /// @dev Flags contract to raise/lower flags on, during status transitions
+  FlagsInterface public immutable FLAGS;
+  /// @dev L1 address
+  address private s_l1Owner;
+  /// @dev Contract initialization flag
+  bool public s_initialized = false;
+
+  uint80 private s_latestRoundId = 0;
+  mapping(uint80 => Round) private s_rounds;
+
+  constructor(address flagsAddress, address l1OwnerAddress) {
+    setL1Owner(l1OwnerAddress);
+
+    FLAGS = FlagsInterface(flagsAddress);
+  }
+
+  /**
+   * @notice Initialise the first round. Can't be done in the constructor,
+   *    because this contract's address must be set as the controller in
+   *    the Flags contract.
+   */
+  function initialize() external onlyOwner {
+    require(!s_initialized, "Already initialised");
+
+    uint64 timestamp = uint64(block.timestamp);
+    bool currentStatus = FLAGS.getFlag(FLAG_ARBITRUM_SEQ_OFFLINE);
+    Round memory initialRound = Round(currentStatus, timestamp);
+    s_rounds[0] = initialRound;
+
+    s_initialized = true;
+    emit Initialized();
+    emit NewRound(0, msg.sender, timestamp);
+    emit AnswerUpdated(getStatusAnswer(initialRound.status), 0, timestamp);
+  }
+
+  /**
+   * @notice versions:
+   *
+   * - Flags 1.0.0: initial release
+   *
+   * @inheritdoc TypeAndVersionInterface
+   */
+  function typeAndVersion() external pure virtual override returns (string memory) {
+    return "ArbitrumSequencerStatusRecorder 1.0.0";
+  }
+
+  /// @return L1 owner address
+  function l1Owner() public view virtual returns (address) {
```

Comment:
Can we call this role something other than "owner"? It doesn't seem to have any real ownership capabilities. Suggestion: `L1 Sender`.

Reply:
Agreed, also slightly confusing when there is also a `ConfirmedOwner`.